### PR TITLE
cfb: introduce invert area function

### PR DESCRIPTION
**Pull request by mzagrabski**
Allow user to invert part of a screen, as a part of rendering.

Signed-off-by: Maciej Zagrabski <maciej.zagrabski@grinn-global.com>

### CODE REVIEW

File: subsys/fb/cfb.c [hunk 1]
```diff
@@ -150,6 +150,41 @@ int cfb_print(const struct device *dev, char *str, uint16_t x, uint16_t y)
...
+		    uint16_t width, uint16_t height)
+{
+	const struct char_framebuffer *fb = &char_fb;
+
+	if ((fb->screen_info & SCREEN_INFO_MONO_VTILED) && !(y % 8)) {
+		if ((x >= fb->x_res) || (y >= fb->y_res)) {
+			LOG_ERR("Coordinates outside of framebuffer");
+
+			return -1;
+		}
```
Comment (jfischer-no) [inline subsys/fb/cfb.c:163]: That is independent of `fb->screen_info`, you can move it up.
Please `return -EINVAL;`, you can fix existing `return -1;` in appropriate places to be consistent, but if please do it in separate commit.

File: subsys/fb/cfb.c [hunk 2]
```diff
@@ -150,6 +150,41 @@ int cfb_print(const struct device *dev, char *str, uint16_t x, uint16_t y)
 	return -1;
 }
 
+int cfb_invert_area(const struct device *dev, uint16_t x, uint16_t y,
+		    uint16_t width, uint16_t height)
+{
+	const struct char_framebuffer *fb = &char_fb;
+
+	if ((fb->screen_info & SCREEN_INFO_MONO_VTILED) && !(y % 8)) {
+		if ((x >= fb->x_res) || (y >= fb->y_res)) {
```
Comment (jfischer-no) [inline subsys/fb/cfb.c:159]: Unnecessary round brackets.

File: subsys/fb/cfb.c (hunk 1 above)
Comment (mzagrabski) [reply subsys/fb/cfb.c]: Return values replace in all places, added as preceding commit. :+1:

**Review by anangl — APPROVED** [review]
LGTM

**Review by jfischer-no — APPROVED** [review]
